### PR TITLE
refactor(Auth): Register and signIn for username/password login

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 1]
```diff
@@ -1,6 +1,41 @@
...
+Auth category can be used to register a user into the auth plugin and sign-in the registered user using username and password in your app. AWS Cognito Auth Plugin is set up with AWS Cognito User Pools that manages the users and their properties.
 
-<inline-fragment platform="ios" src="~/lib/auth/fragments/native_common/authentication/common.md"></inline-fragment>
-<inline-fragment platform="android" src="~/lib/auth/fragments/native_common/authentication/common.md"></inline-fragment>
+## Prerequisites
+* An iOS application targeting at least iOS 11.0
+* Integrated with Amplify Auth and ability to sign in with an account
+  * For a full example, see: [Auth Getting Started](~/lib/auth/getting-started.md)
+
+## Configure the Backend
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:9]: I don't believe we're using Title Case for these, so:

## Configure the backend

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 2]
```diff
@@ -1,6 +1,41 @@
 
-<inline-fragment platform="ios" src="~/lib/auth/fragments/native_common/authentication/register_user/common.md"></inline-fragment>
-<inline-fragment platform="android" src="~/lib/auth/fragments/native_common/authentication/register_user/common.md"></inline-fragment>
+Auth category can be used to register a user into the auth plugin and sign-in the registered user using username and password in your app. AWS Cognito Auth Plugin is set up with AWS Cognito User Pools that manages the users and their properties.
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:2]: Hmm. This graf reads oddly. What are we trying to express here? Let me give it a shot:

> The Authentication category can be used to manage your user directory using Amazon Cognito User Pools. Your users can register and authenticate against a User Pool which will store things like their password, email, and other properties.

Be careful with service names -- it's Amazon Cognito, Not AWS Cognito.

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 3]
```diff
@@ -1,6 +1,41 @@
...
 
-<inline-fragment platform="ios" src="~/lib/auth/fragments/native_common/authentication/common.md"></inline-fragment>
-<inline-fragment platform="android" src="~/lib/auth/fragments/native_common/authentication/common.md"></inline-fragment>
+## Prerequisites
+* An iOS application targeting at least iOS 11.0
+* Integrated with Amplify Auth and ability to sign in with an account
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:6]: "and ability to sign in with an account"

As a reader, I'm not sure what this means.

Also, we should standardize: Auth or Authentication?

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 4]
```diff
@@ -1,6 +1,41 @@
...
+  * For a full example, see: [Auth Getting Started](~/lib/auth/getting-started.md)
+
+## Configure the Backend
+
+The backend will be already configured if you have followed the basic setup as mentioned in the prerequisites.
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:11]: Link? Wait is this all of the contents of this whole subsection?

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 5]
```diff
@@ -1,6 +1,41 @@
...
+
+## Register a user
+
+A user should exists in the backend to move forward with the next step of authentication. The defaul CLI flow require a username, password and a valid email id as parameters to register a user. Create a UI to get these values from the user. When you have the needed details invoke the following function to initiate a signUp flow.
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:15]: Not sure what the first sentence means: suggest deleting it.

Comment (jpignata) [reply docs/lib/auth/fragments/native_common/authentication/signin/common.md:15]: We should have a defined CLI flow somewhere, and walk the user through it. "The default CLI flow" means nothing to a reader.

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 6]
```diff
@@ -1,6 +1,41 @@
...
+
+<inline-fragment platform="ios" src="~/lib/auth/fragments/ios/authentication/signin/10_signUp.md"></inline-fragment>
+
+If you have followed the default setup in the CLI, the next step for signUp flow will be to confirm the user. A confirmation code will be send to the email id that was given during the above step. Enter the confirmation code received in the email in the next step.
+
+<inline-fragment platform="ios" src="~/lib/auth/fragments/ios/authentication/signin/20_confirmSignUp.md"></inline-fragment>
+
+After this step signUp flow is complete if you see the following in your console window:
+
+```bash
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:25]: This isn't `bash`, it's `console`.

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 7]
```diff
@@ -1,6 +1,41 @@
...
+
+The backend will be already configured if you have followed the basic setup as mentioned in the prerequisites.
+
+## Register a user
+
+A user should exists in the backend to move forward with the next step of authentication. The defaul CLI flow require a username, password and a valid email id as parameters to register a user. Create a UI to get these values from the user. When you have the needed details invoke the following function to initiate a signUp flow.
+
+<inline-fragment platform="ios" src="~/lib/auth/fragments/ios/authentication/signin/10_signUp.md"></inline-fragment>
+
+If you have followed the default setup in the CLI, the next step for signUp flow will be to confirm the user. A confirmation code will be send to the email id that was given during the above step. Enter the confirmation code received in the email in the next step.
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:19]: Same comment as above.

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 8]
```diff
@@ -1,6 +1,41 @@
...
+
+## Sign in a user 
+
+Implement an UI to get the username and password from the user. After the user enter the username and password you can start the signIn flow by calling the following method:
+
+<inline-fragment platform="ios" src="~/lib/auth/fragments/ios/authentication/signin/30_signIn.md"></inline-fragment>
+
+After this step signIn flow is complete and you will see the following in your console window:
+
+```bash
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:37]: `console`

File: docs/lib/auth/fragments/native_common/authentication/signin/common.md [hunk 9]
```diff
@@ -1,6 +1,41 @@
...
+Sign in succeeded
+```
+
+Congratulations!! you have sucessfully registered and authenticated a user into the auth category of Amplify. If you would like to explore other ways of authenticating a user, check out the next sections.
```
Comment (jpignata) [inline docs/lib/auth/fragments/native_common/authentication/signin/common.md:41]: Suggestion:

You have successfully registered a user and authenticated with that user's username and password using Amplify. The Authentication category supports other mechanisms for authentication such as whatever, something-else, and-the-other-thing that you can explore below.

## Next steps

- Thing1
- Thing2
- Thing3

**Comment by jpignata** [comment]
Nit: the token for type should be "docs", not "refactor"

**Comment by TrekSoft** [comment]
@royjit - let's just do pushes against this branch for now and then do a final PR from Auth to Next. You can just implement these changes John suggested.

**Pull request merged**